BitmarkRegistry/Services/ChibitronicsService.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  ChibitronicsService.swift
+//  BitmarkRegistry
+//
+//  Created by Macintosh on 8/1/19.
+//  Copyright © 2019 thuyentruong. All rights reserved.
+//
+
+import Foundation
+import BitmarkSDK
+
+enum VerificationLinkSource {
+  case qrCode, deepLink
+}
+
+struct ChibitronicsService {
+
+  let verificationLink: String
+  let source: VerificationLinkSource
+  lazy var separator: Character = { source == .qrCode ? "|" : "/" }()
+  lazy var regex: String = { "(\\w+ ){5}\\w+\\\(separator)https?://.+" }()
+
+  var code: String!
+  var url: URL!
+
+  init(verificationLink: String, source: VerificationLinkSource) {
+    self.verificationLink = verificationLink
+    self.source = source
+  }
+
+  mutating func isValid() -> Bool {
+    return verificationLink.range(of: regex, options: .regularExpression, range: nil, locale: nil) != nil
+  }
+
+  mutating func extractData() -> (code: String, url: URL)? {
+    let codeParts = verificationLink.split(separator: separator, maxSplits: 1)
+
+    if let url = URL(string: String(codeParts[1])) {
+      self.code = String(codeParts[0])
+      self.url = url
+      return (code: code, url: url)
+    } else {
+      return nil
+    }
+  }
+
+  func requestAuthorization(for account: Account, completion: @escaping (Error?) -> Void) throws {
```

Comment:
Typically we can leave `completion` as optional with `nil` by default, unless it serves for special purposes.

Reply:
Sorry, after check again, I think in this case the completion handler is required;
when it's error => show error alert
otherwise        => show success alert and process authorized success process